core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/Workflow.scala

```diff
@@ -253,13 +253,22 @@ class Workflow(
     val sender = from._2
     val receiver = to._2
     val receiverOpExecConfig = to._1
-    if (receiverOpExecConfig.requiredShuffle) {
+    if (receiverOpExecConfig.requiresHashBasedShuffle) {
```

Comment:
I suggest we use an enum for shuffle type here.

Reply:
done